modules/solid_mechanics/src/vectorpostprocessors/CrackFrontNonlocalMaterialBase.C

```diff
@@ -0,0 +1,148 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CrackFrontNonlocalMaterialBase.h"
+#include "CrackFrontDefinition.h"
+
+InputParameters
+CrackFrontNonlocalMaterialBase::validParams()
+{
+  InputParameters params = ElementVectorPostprocessor::validParams();
+  params.addRequiredParam<UserObjectName>("crack_front_definition",
+                                          "The CrackFrontDefinition user object name");
+  params.addRequiredParam<Real>("box_length", "Distance in front of crack front.");
+  params.addRequiredParam<Real>("box_height", "Distance normal to front of crack front.");
```

Comment:
```suggestion
  params.addRequiredParam<Real>("box_length", "Dimension of property-averaging box in direction of crack extension.");
  params.addRequiredParam<Real>("box_height", "Dimension of property-averaging box in direction normal to crack.");
```

Reply:
I should note that the terminology of the "normal" direction is kind of ambiguous. I think we use that pretty widely so I'm OK with it as long as it's documented somewhere. Do we explain that anywhere in the documentation?